scripts/docker-helper/cht-docker-compose.sh

```diff
@@ -0,0 +1,442 @@
+  for container in "${containersArray[@]}"
+  do
+    if [ "$( docker ps -f name="${container}" | wc -l )" -eq 2 ]; then
+        (( result++ ))
+    fi
+  done
+  echo "$result"
+}
+
+get_global_running_container_count(){
+  # shellcheck disable=SC1083
+  if [ "$( docker ps --format={{.Names}} | wc -l )" -gt 0 ]; then
+    docker ps --format={{.Names}} | wc -l
+  else
+    echo "0"
+  fi
+}
+
+volume_exists(){
+  project=$1
+  volume="${project}_medic-data"
+  if [ "$( docker volume inspect "${volume}" 2>&1  | wc -l )" -eq 2 ]; then
+    echo "0"
+  else
+    echo "1"
+  fi
+}
+
+get_images_count(){
+  images="$1"
+  result=0
+  IFS=' ' read -ra imagesArray <<< "$images"
+  for image in "${imagesArray[@]}"
+  do
+    if [ "$( docker image ls --format {{.Repository}}:{{.Tag}} | grep -c "${image}" )" -eq 1 ]; then
+        (( result++ ))
+    fi
+  done
+  echo "$result"
+}
+
+pull_images(){
+  images="$1"
+  IFS=' ' read -ra imagesArray <<< "$images"
+  for image in "${imagesArray[@]}"
+  do
+    docker image pull "${image}" >/dev/null 2>&1
+  done
+}
+
+get_docker_compose_yml_path(){
+  if [ -f docker-compose-developer.yml ]; then
+    echo "docker-compose-developer.yml"
+  elif [ -f ../../docker-compose-developer.yml ]; then
+    echo "../../docker-compose-developer.yml"
+  else
+    return 0
+  fi
+}
+
+docker_up_or_restart(){
+  # some times this function called too quickly after a docker change, so
+  # we sleep 3 secs here to let the docker container/volume stabilize
+  sleep 3
+
+  envFile=$1
+  composeFile=$2
+
+  # haproxy never starts on first "up" call, so you know, call it twice ;)
+  docker-compose --env-file "${envFile}" -f "${composeFile}" down >/dev/null 2>&1
+  docker-compose --env-file "${envFile}" -f "${composeFile}" up -d >/dev/null 2>&1
+  docker-compose --env-file "${envFile}" -f "${composeFile}" up -d >/dev/null 2>&1
+}
+
+install_local_ip_cert(){
+  medicOs=$1
+  docker exec -it "${medicOs}" bash -c "curl -s -o server.pem http://local-ip.co/cert/server.pem" >/dev/null 2>&1
+  docker exec -it "${medicOs}" bash -c "curl -s -o chain.pem http://local-ip.co/cert/chain.pem" >/dev/null 2>&1
+  docker exec -it "${medicOs}" bash -c "cat server.pem chain.pem > /srv/settings/medic-core/nginx/private/default.crt" >/dev/null 2>&1
+  docker exec -it "${medicOs}" bash -c "curl -s -o /srv/settings/medic-core/nginx/private/default.key http://local-ip.co/cert/server.key" >/dev/null 2>&1
+  docker exec -it "${medicOs}" bash -c "/boot/svc-restart medic-core nginx" >/dev/null 2>&1
+}
+
+docker_down(){
+  envFile=$1
+  composeFile=$2
+  docker-compose --env-file "${envFile}" -f "${composeFile}" down >/dev/null 2>&1
+}
+
+docker_destroy(){
+  project=$1
+  containers=$2
+  IFS=' ' read -ra containersArray <<<"$containers"
+  for container in "${containersArray[@]}"; do
+    docker stop -t 0 "${container}" >/dev/null 2>&1
+  done
+  docker rm "${project}"_haproxy_1 "${project}"_medic-os_1 >/dev/null 2>&1
```

Comment:
`docker-compose down` does most of this already

Reply:
Agreed!  And you could even follow [some of our existing tips](https://docs.communityhealthtoolkit.org/core/guides/docker-setup/#delete--re-install) on how to do this.  However, I'm trying to make this helper script **The One And Only One** you run once to set `up` your app development environment and then when you're done, you can pull it `down`.  

Breaking them out into [raw `docker-compose` CLI has failed already](https://github.com/medic/cht-core/issues/6456), and we're looking for an improvement.